plugins/src/android/OptimoveSDKPlugin.java

```diff
@@ -138,4 +320,142 @@ private void getCurrentUserIdentifier(CallbackContext callbackContext) {
             callbackContext.error("current user identifier is null");
         }
     }
+
+    private void setHandlersCallBackContext(CallbackContext callbackContext) {
+        jsCallbackContext = callbackContext;
+        PluginResult result = new PluginResult(PluginResult.Status.OK);
+        result.setKeepCallback(true);
```

Comment:
just a question, what does this do?

Reply:
this function save a reference to a callback context in which through it we send messages back to the js (like receiving an event in the java code and send it to a handler in the js code)

Reply:
i was asking more about this `result.setKeepCallback(true);`

Reply:
this function keeps the contextCallback "alive" so we can use it multiple times

Reply:
https://ourcodeworld.com/articles/read/33/how-to-execute-a-cordova-callback-more-than-once-in-java-android-

Reply:
👍

Reply:
Ok, does `OptimoveSDKPlugin.jsCallbackContext.error(...` invalidate the context then? based on this it does 

https://github.com/apache/cordova-android/blob/master/framework/src/org/apache/cordova/CallbackContext.java#L121

Reply:
@vvoicehovics sorry think perhaps the line is messed up, do you mean to do with the setting of the `finished` field on the context in `sendPluginResult` because it is a new result instance?

So to resolve that we should create a new PluginResult ourselves & set to keep context in any error case?

Maybe it doesn't matter, are there any non-terminal error states that could be trapped?

Reply:
I think reusing `OptimoveSDKPlugin.jsCallbackContext.error()` is not going to work because

1) yes, when we call that it does 

```
  public void error(JSONObject message) {
        sendPluginResult(new PluginResult(PluginResult.Status.ERROR, message));
    }
```
(without setKeepCallback). This would invalidate context. We could work around, but mb we just avoid using `OptimoveSDKPlugin.jsCallbackContext...`. because
2a) i dont kno wto which callback it would report error
2b) we dont really need to report errors there

soooo, propose to just only use `OptimoveSDKPlugin.jsCallbackContext` for handlers

Reply:
we eventually used sendPluginResult to report the error which doesn't invalidate the context